ADR/0008-integration-service-api-contracts.md

```diff
@@ -0,0 +1,277 @@
+
+- Parameter names: dash-case (already used in build Tasks)
+- Resource names: As per [Kubernetes recommendations](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/) resource names will follow  [RFC 1123](https://datatracker.ietf.org/doc/html/rfc1123)
+- Environment variables: Following [Kubernetes](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/) and bash practices environment variables should be defined in UPPER_CASE
+- Labels: Labels names will follow Kubernetes practices: prefix with dot `.` as separator and dash-case for names. Prefixes and names are separated by a slash `/`. In addition to the [labels automatically created by Tekton Kubernetes standard labels or custom labels](https://tekton.dev/vault/pipelines-v0.14.3/labels/#automatically-added-labels) could be leveraged to further qualify the resources. Possible groupings (TODO: if desired, define common names):
+  - Pipeline part: Build, Test, Release, etc.
+  - Application/Product being built: AMQ Streams, EAP, RHSSO, etc.
+
+### DRY
+
+Information, which relates to an environment like (7) connection details and credentials should be configured once at the environment level and not passed as parameters for every PipelineRun. On the other hand, information, which is specific to a run like a container image digest, may be passed as a parameter.
+
+### Information injection
+
+Whenever possible, resources like ConfigMaps or Secrets will be used to inject configuration into Tasks. This is preferred to templates and patches as it fits well with Kubernetes declarative and GitOps approaches.
+
+ConfigMaps and Secrets will be mounted into the Task pod to inject file-based information like certificates.
+Environment variables may be injected from ConfigMaps.
+
+Variables directly configured in the Pod definition are discouraged.
+
+Side note: Secrets and ConfigMaps should be made [immutable](https://kubernetes.io/docs/concepts/configuration/secret/#secret-immutable)
+
+### Information vector
+
+When to use:
+- Workspace
+- ConfigMap / Secret mount
+- Injection of variable from ConfigMap
+- Volumes
+- Task Parameters
+- Task Results
+- Tekton Results
+- PipelineResources (resources input / output)  are deprecated: don’t use
+
+
+The maximum size of a Task's Results is limited by the container termination message feature of Kubernetes.
+
+
+App Studio builds are structured as [a shared Persistent Volume per App Studio Workspace](https://docs.google.com/document/d/1IPlihVjkJ4Kb9tdhsk7iz3bn5rkT_SvJCNQhyXzK3aI/edit#). This allows teams to share builds, implement caching and other shared volumes. A single persistent volume is mapped to each default build pipeline. Builds are passed a directory specific to their builds.
```

Comment:
The link point to the header of the document, can you please change it to point to the section about storage?
In addition, this repository is public, so we need to attach only documents that are public.

Reply:
Thanks for you advice. I will change it to point to the section about storage.

About attaching public doc, do you have any idea if we want to link that g-doc to make the users of this ADR know it? Maybe we can use this g-doc link and then update the link once the doc from build team is made public in the future?

Reply:
The way I'm thinking about it now is that I'd rather link to a private doc than not link at all. As we open up this doc and others, over time we should wind up with only public-facing links. The ADRs are historical documents, documenting decisions at a point in time. The reality today is that most of those docs are private, but here we are working to change that, one document at a time. :)